### DISCUSSION TITLE
Templating reference

### DISCUSSION THREAD

**Discussion by paraddise** [secureCodeBox/secureCodeBox#2707] [Q&A] [answered]
I'm trying to create versioned engagement in DefectDojo with ScheduledScan.  I didn't managed to find reference to configure templating in annotations. Can you say what fields I can reference in annotations or place in the code responsible for templating?

**Comment by J12934** [comment]
Hi, sorry I don't get the question
The DefectDojo doesn't template anything. 

The annotations used by the hook are documented here: https://www.securecodebox.io/docs/hooks/defectdojo#additional-chart-configurations.

Do you mean the templating by either the auto discovery or from cascading scans?

**Reply by paraddise** [reply]
Yes, I mean templating by auto discovery. I want to set engagement version from time of scan start

**Reply by paraddise** [reply]
@J12934 
> Do you mean the templating by either the auto discovery or from cascading scans?

yes, can you tell more about this templating

**Comment by J12934** [comment, marked as answer]
The auto-discovery templating params are right now not really nicely documented.
Easiest reference are the example templates configured by default: https://github.com/secureCodeBox/secureCodeBox/blob/main/auto-discovery/kubernetes/values.yaml#L43

For more details you can reference the parameters handed intot he templating function in code.
These can be found here:

- For Container AutoDiscovery: https://github.com/secureCodeBox/secureCodeBox/blob/main/auto-discovery/kubernetes/controllers/container_scan_controller.go#L240
- For Service AutoDiscovery: https://github.com/secureCodeBox/secureCodeBox/blob/main/auto-discovery/kubernetes/controllers/service_scan_controller.go#L149
